sematic/resolvers/cloud_resolver.py

```diff
@@ -261,7 +258,7 @@ def _process_run_output(self, run_id: str) -> None:
             return
 
         if run.nested_future_id is not None:
-            pickled_nested_future = self._storage.get(
+            pickled_nested_future = S3Storage().get(
```

Comment:
I assume this PR can't be released until this functionality is propagated to nested futures, correct?

Reply:
It can. Nested futures still use the legacy `S3Storage` class, as shown here. This PR does not remove this class, it simply migrates artifacts to use the new storage plug-ins.